uber/model_checks.py

```diff
@@ -448,8 +449,19 @@ def valid_uses_allowed(promo_code):
 
 
 @validation.PromoCode
+def no_unlimited_free_badges(promo_code):
+    if promo_code.is_unlimited and (promo_code.is_free or (
+            promo_code.discount_type == PromoCode.FIXED_DISCOUNT and
+            promo_code.discount >= c.BADGE_PRICE)):
+        return 'Unlimited-use, free-badge promo codes are not allowed.'
```

Comment:
IMO we shouldn't give away free badges in this scenario - I think it is perfectly valid to effectively say "Promo codes worth more than the current badge price are not valid."